### PR TITLE
Use java11 for builds

### PR DESCRIPTION
**Pull request by Xenorith**
update default alluxio-maven docker image used to the `0.1.0-jdk11` tag. this changes the default jdk from 8 to 11 and upgrades maven from 3.6.3 to 3.8.5

rename github action workflows from jdk8 -> jdk11

### CODE REVIEW

File: dev/github/run_checks.sh [hunk 1]
```diff
@@ -28,10 +30,10 @@ fi
 
 export MAVEN_OPTS="-Dorg.slf4j.simpleLogger.showDateTime=true -Dorg.slf4j.simpleLogger.dateTimeFormat=HH:mm:ss.SSS"
 
-# Always use java 8 to compile the source code
+# Always use java 11 to compile the source code
```
Comment (ssz1997) [inline dev/github/run_checks.sh:33]: I think we had this line of command to note that although we are running tests in java 11, we still built with java 8. Now if everything is 11, this comment doesn't serve any purpose.

Comment (Xenorith) [reply dev/github/run_checks.sh:33]: we may introduce a java17 build, so i left these extraneous comments and no-op settings

Comment (ssz1997) [reply dev/github/run_checks.sh:33]: ok

**Review by beinan — APPROVED** [review]
lgtm

**Comment by beinan** [comment]
alluxio-bot, merge this please.

**Comment by Xenorith** [comment]
alluxio-bot, merge this please

**Pull request merged**